hybrid/per_func_comp_main.c

```diff
@@ -0,0 +1,257 @@
+
+    //- Create cheri_object_1 *code* cap -//
+
+    void *sandbox_code_1_ubound = (void *) (((size_t) sandbox_1_func) + SANDBOX_CODE_SZ);
+    cheriobj_1_codecap = cheri_seal(
+        codecap_create((void (*)(void)) & sandbox_1_func, sandbox_code_1_ubound), sealcap_1);
+
+    //- Create cheri_object_1 *data* cap -//
+
+    //- Two cheri_objects' data addr ranges are set overlapped. Ignore the numbers for this example.
+    //-//
+    void *apprx_data_UB = __builtin_frame_address(0);
+
+    size_t cheriobj_data_1_base = (size_t) apprx_data_UB - 1000; // approx
+    size_t cheriobj_data_1_end = (size_t) apprx_data_UB;
+    cheriobj_1_datacap = cheri_seal(
+        datacap_create((void *) cheriobj_data_1_base, (void *) cheriobj_data_1_end), sealcap_1);
+
+    //=- Setup cheri_object_2 -=//
+
+    //- Create cheri_object_2 sealcap -//
+
+    size_t sealcap_size_2 = 32; // arbitrary type size for sealcap2
+    void *__capability sealcap_2 = alloc_type(sealcap_size_2);
+
+    //- Create cheri_object_2 *code* cap -//
+
+    void *cheriobj_2_code_base = (void *) (&sandbox_2_func);
+    void *cheriobj_2_code_end = (void *) (((size_t) cheriobj_2_code_base) + SANDBOX_CODE_SZ);
+    cheriobj_2_codecap = cheri_seal(
+        codecap_create((void (*)(void))(&sandbox_2_func), cheriobj_2_code_end), sealcap_2);
+
+    assert((size_t) &sandbox_1_func != (size_t) &sandbox_2_func);
+
+    //- Create cheri_object_2 *data* cap -//
+    void *cheriobj_2_data_base = (void *) ((size_t) apprx_data_UB - 1000); // approx
+    cheriobj_2_datacap = cheri_seal(datacap_create(cheriobj_2_data_base, apprx_data_UB), sealcap_2);
+
+    // Check
+    assert(cheri_is_sealed(cheriobj_1_codecap));
+    assert(cheri_is_sealed(cheriobj_1_datacap));
+    assert(cheri_is_sealed(cheriobj_2_codecap));
+    assert(cheri_is_sealed(cheriobj_2_datacap));
+
+    printf("____global_sealcap__________\n");
+    print_cap(global_sealcap);
+
+    printf(">>>> cheri_object_1__________________________\n");
+    printf("Sealcap_1_user_defined_tysz : %04lx\n", sealcap_size_1);
+    printf("___sealcap_1_info____\n");
+    print_cap(sealcap_1);
+    printf("___code_cap_1_sealed_info____\n");
+    print_cap(cheriobj_1_codecap);
+    printf("___data_cap_1_sealed_info____\n");
+    print_cap(cheriobj_1_datacap);
+
+    printf(">>>> cheri_object_2__________________________\n");
+
+    printf("Sealcap_2_user_defined_tysz : %04lx\n", sealcap_size_2);
+    printf("___sealcap_2_info____\n");
+    print_cap(sealcap_2);
+    printf("___code_cap_2_sealed_info____\n");
+    print_cap(cheriobj_2_codecap);
+    printf("___data_cap_2_sealed_info____\n");
+    print_cap(cheriobj_2_datacap);
+
+    printf(" ** code_1_type: %04x\n", cheri_type_get(cheriobj_1_codecap));
+    printf(" ** data_1_type: %04x\n", cheri_type_get(cheriobj_1_datacap));
+    printf(" ** code_2_type: %04x\n", cheri_type_get(cheriobj_2_codecap));
+    printf(" ** data_2_type: %04x\n\n", cheri_type_get(cheriobj_2_datacap));
```

Comment:
Do we check anything with these prints? If so, I would convert them to `assert`s, otherwise I would remove them.

Reply:
Just to show values at one glance. I wouldn't remove them for myself.

Reply:
I think Andrei's question is: could these be `assert`s? That way they're checked automatically, rather than us having to inspect them visually.